common/Freemium.cpp

```diff
@@ -0,0 +1,44 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+#include <string>
+#include <vector>
+#include "ConfigUtil.hpp"
+#include "Util.hpp"
+#include "Freemium.hpp"
+
+#include <boost/algorithm/string/trim.hpp>
+
+namespace Freemium {
+    std::vector<std::string> FreemiumManager::FreemiumBlacklist;
+
+    FreemiumManager::FreemiumManager() {}
+
+    const std::vector<std::string>& FreemiumManager::getFreemiumBlacklist()
+    {
+#ifdef ENABLE_FREEMIUM
+        // Return if array is already filled
+        if (FreemiumBlacklist.size() != 0)
+            return FreemiumBlacklist;
+
+        StringVector commandList;
+        if (config::getString("freemium.disabled_commands", "") != "")
+            commandList = Util::tokenize(config::getString("freemium.disabled_commands", ""), "\n");
+
+        std::string command;
+        for (std::size_t i = 0; i < commandList.size(); i++)
```

Comment:
No need to index through it, range-for works perfectly here.

```
for (const auto& command : commandList)
    FreemiumBlacklist.emplace_back(Util::trimmed(command));
```

Reply:
it would not work with StringVector. The name is confusing. But String vector is not a vector and iterators does not return string but StringToken which is just integers for position and size.